todos/src/redux/todoSlice.jsx

```diff
@@ -0,0 +1,27 @@
+import { createSlice } from "@reduxjs/toolkit";
+
+export const todoSlice = createSlice({
+  name: "todos",
+  initialState: [],
+  reducers: {
+    addTodo: (state, action) => {
+      state.push({
+        id: state.length + 1,
+        todo: action.payload,
+        hasEnded: false,
+      });
+    },
+    endTask: (state, action) => {
+      const todo = state[action.payload];
+      if (todo) {
+        todo.hasEnded = !todo.hasEnded;
+      }
+    },
```

Comment:
According to the implementation on this line `todo.hasEnded = !todo.hasEnded;` this reducer will be used to end a todo when it's not ended but will also be used to unEnd it when it's already ended. Naming it `endTask` only mentions one use-case. It would be better to give a name that  Therefore its name should reflect both usages. 

By the way, It would be better to stick to one keyword in your naming and not mix `todo` and `task` currently you have `deleteTask`, `endTask` but also `addTodo` they should be all named with one keyword for consistency

Reply:
Done